scripts/run_e2e_tests.py

```diff
@@ -222,397 +117,172 @@ def is_oppia_server_already_running():
-            'This is used to determine the chromedriver version to use. '
-            'Please set the chromedriver version manually using '
-            '--chrome_driver_version flag. To determine the chromedriver '
-            'version to be used, please follow the instructions mentioned '
-            'in the following URL:\n'
-            'https://chromedriver.chromium.org/downloads/version-selection' % (
-                ' '.join(arg.replace(' ', r'\ ') for arg in popen_args)
-            )
-        )
-    chrome_version = ''.join(re.findall(r'([0-9]|\.)', output))
-    chrome_version = '.'.join(chrome_version.split('.')[:-1])
-    response = python_utils.url_open(
-        'https://chromedriver.storage.googleapis.com/LATEST_RELEASE_%s'
-        % chrome_version)
-    chrome_driver_version = response.read()
-    python_utils.PRINT('\n\nCHROME VERSION: %s' % chrome_version)
-    return chrome_driver_version
-
-
-def start_portserver():
-    """Start a portserver in a subprocess.
-
-    The portserver listens at PORTSERVER_SOCKET_FILEPATH and allocates free
-    ports to clients. This prevents race conditions when two clients
-    request ports in quick succession. The local Google App Engine
-    server that we use to serve the development version of Oppia uses
-    python_portpicker, which is compatible with the portserver this
-    function starts, to request ports.
-
-    By "compatible" we mean that python_portpicker requests a port by
-    sending a request consisting of the PID of the requesting process
-    and expects a response consisting of the allocated port number. This
-    is the interface provided by this portserver.
-
-    Returns:
-        subprocess.Popen. The Popen subprocess object.
-    """
-    process = subprocess.Popen([
-        'python', '-m',
-        '.'.join(['scripts', 'run_portserver']),
-        '--portserver_unix_socket_address',
-        PORTSERVER_SOCKET_FILEPATH,
-    ])
-    return process
-
-
-def cleanup_portserver(portserver_process):
-    """Shut down the portserver.
-
-    We wait KILL_TIMEOUT_SECS seconds for the portserver to shut down
-    after sending CTRL-C (SIGINT). The portserver is configured to shut
-    down cleanly upon receiving this signal. If the server fails to shut
-    down, we kill the process.
-
-    Args:
-        portserver_process: subprocess.Popen. The Popen subprocess
-            object for the portserver.
-    """
-    _kill_process(portserver_process)
-
-
 def run_tests(args):
     """Run the scripts to start end-to-end tests."""
-    oppia_instance_is_already_running = is_oppia_server_already_running()
+    # TODO(#11549): Move this to top of the file.
+    import contextlib2
 
-    if oppia_instance_is_already_running:
+    if is_oppia_server_already_running():
         sys.exit(1)
-    setup_and_install_dependencies(args.skip_install)
 
-    common.start_redis_server()
-    atexit.register(cleanup)
+    install_third_party_libraries(args.skip_install)
```

Comment:
Why don't we just do:

```python
if not args.skip_install:
    install_third_party_libraries()
```

That seems clearer to me

Reply:
I believe it's to make the unit tests easier to mock (this is how I found it before my changes).

Reply:
ah okay